### PR TITLE
Minor typo fixes

### PR DESCRIPTION
**Pull request by sam-bristow**
No functional changes to the codebase.

Automatically corrected with [codespell](https://github.com/lucasdemarchi/codespell).

Signed-off-by: Sam Bristow <sam.bristow@gmail.com>

### CODE REVIEW

File: docs/source/release-notes.rst [hunk 1]
```diff
@@ -42,7 +42,7 @@ were an attempt to make a forward-looking guarantee that was entirely unneeded.
 
 Altogether, this has been an extremely productive period for Hyper-h2, and a
 lot of great work has been done by the community. To that end, we'd also like
-to extend a great thankyou to those contributers who made their first contribution
+to extend a great thankyou to those contributors who made their first contribution
```
Comment (nateprewitt) [inline docs/source/release-notes.rst]: Might as well get "thankyou" while we're at it ;)

Comment (sam-bristow) [inline docs/source/release-notes.rst]: I thought about that. At least for non-US English the noun 'thankyou' is one word. [\[ref\]](http://www.future-perfect.co.uk/grammar-tip/is-it-thank-you-thankyou-or-thank-you/).

Comment (nateprewitt) [inline docs/source/release-notes.rst]: Huh, you learn something new everyday. Thanks for the education :)

Comment (sam-bristow) [inline docs/source/release-notes.rst]: Yeah, I only know that because I looked it up when the tool didn't flag it. :-)

**Comment by sam-bristow** [comment]
I'm a bit confused, did this get merged in the wrong direction?

**Comment by Lukasa** [comment]
No, the branch was not up-to-date with master and I try not to merge when that's the case. So I brought it up-to-date with master and let the tests run again.

**Pull request merged**